plugins/CuraEngineBackend/CuraEngineBackend.py

```diff
@@ -476,14 +486,14 @@ def _onSceneChanged(self, source: SceneNode) -> None:
         else:
             # we got a single scenenode
             if not source.callDecoration("isGroup"):
-                if source.getMeshData() is None:
-                    return
-                if source.getMeshData().getVertices() is None:
+                meshData = source.getMeshData();
+                if meshData and meshData.getVertices() is None:
                     return
 
             build_plate_changed.add(source_build_plate_number)
 
-        build_plate_changed.discard(None)
+        # TODO: Commented out for code tests, discard() only takes 'int', and no explanation exists for this line
+        # build_plate_changed.discard(None)
```

Comment:
The weird thing is that in the function `_numObjectsPerBuildPlate` if for some reason the node doesn't have the decoration `getBuildPlateNumber`, a `None` value is added to the dict and then added to the `build_plate_changed` set.
I already did this in https://github.com/Ultimaker/Cura/commit/7d7a51d77241026ef665300b944b64dd69dbddcc, but it was reverted completely for some reasons.

Reply:
Reverted, I think, because it caused Cura to complete crash. Better to ask whoever committed the revert.

Reply:
Yes, I guess it was reverted for some major issue. But it still needs to be fixed.